packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -223,12 +224,73 @@ void updateLocalProperties({String projectPath, BuildInfo buildInfo}) {
     settings.writeContents(localProperties);
 }
 
+void findAndReplaceVersionProperties({BuildInfo buildInfo}) {
```

Comment:
This should return `Future<Null>` and use a completer to complete the future once our input stream is done.  Then we should `await` this function in `buildGradleProject()`